### PR TITLE
Add the new sast-snyk-check-oci-ta Task

### PR DESCRIPTION
**Pull request by lcarva**
This pull request introduces a new Tekton Task, `sast-snyk-check-oci-ta`. This is a variation of the `0.1/sast-snyk-check` Task. The main different is that instead of expecting the application source code to come from a workspace, this Task consumes the source code as a Trusted Artifact from an OCI registry.

The Task is not currently used in any of the default Pipelines. This will happen later on.

I tested this Task on a kind cluster like this:

```
# First created a secret with the Snyk token and linked it to my account, then:
🐚 oc -n default apply -f task/sast-snyk-check-oci-ta/0.1/sast-snyk-check-oci-ta.yaml

🐚 tkn -n default task start sast-snyk-check-oci-ta --showlog \
    --param SOURCE_ARTIFACT=oci:quay.io/lucarval/ec-551@sha256:450d6a50de35262296c79860dcc3369a25eec205a39ac4d6921c13763412f103 \
    --use-param-defaults
TaskRun started: sast-snyk-check-oci-ta-run-fsb7w
Waiting for logs to be available...
[use-trusted-artifact] Using token for quay.io
[use-trusted-artifact] Restored artifact quay.io/lucarval/ec-551@sha256:450d6a50de35262296c79860dcc3369a25eec205a39ac4d6921c13763412f103 to /var/workdir/source
[use-trusted-artifact] 

[sast-snyk-check] {
[sast-snyk-check]   "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
[sast-snyk-check]   "version": "2.1.0",
[sast-snyk-check]   "runs": [
[sast-snyk-check]     {
[sast-snyk-check]       "tool": {
[sast-snyk-check]         "driver": {
[sast-snyk-check]           "name": "SnykCode",
[sast-snyk-check]           "semanticVersion": "1.0.0",
[sast-snyk-check]           "version": "1.0.0",
[sast-snyk-check]           "rules": []
[sast-snyk-check]         }
[sast-snyk-check]       },
[sast-snyk-check]       "results": [],
[sast-snyk-check]       "properties": {
[sast-snyk-check]         "coverage": [
[sast-snyk-check]           {
[sast-snyk-check]             "isSupported": true,
[sast-snyk-check]             "lang": "Go",
[sast-snyk-check]             "files": 173,
[sast-snyk-check]             "type": "SUPPORTED"
[sast-snyk-check]           }
[sast-snyk-check]         ]
[sast-snyk-check]       }
[sast-snyk-check]     }
[sast-snyk-check]   ]
[sast-snyk-check] }
[sast-snyk-check] {"result":"SUCCESS","timestamp":"1715719388","note":"For details, check Tekton task log.","namespace":"default","successes":0,"failures":0,"warnings":0}
```

Resolves [EC-553](https://issues.redhat.com/browse/EC-553)

### CODE REVIEW

**Comment by lcarva** [comment]
Tip for reviewers: the first commit in this PR is a plain copy of the `0.1/sast-snyk-check` Task. This should make it easier to identify the actual changes in the new Task.

File: task/sast-snyk-check-oci-ta/0.1/README.md [hunk 1]
```diff
@@ -1,35 +1,22 @@
...
 
-The sast-snyk-check task uses Snyk Code tool to perform Static Application Security Testing (SAST) for Snyk, a popular cloud-native application security platform.
+Follow the steps given [here](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/testing_applications/enable_snyk_check_for_a_product/) to obtain a snyk-token and to enable the snyk task in a Pipeline.
 
-Snyk's SAST tool uses a combination of static analysis and machine learning techniques to scan an application's source code for potential security vulnerabilities, including common issues such as SQL injection, cross-site scripting (XSS), and code injection attacks.
+The snyk binary used in this Task comes from a container image defined in https://github.com/konflux-ci/konflux-test
 
-> NOTE: This task is executed only if the user provides a Snyk token stored in a secret in their namespace. The name of the secret then needs to be supplied in the `snyk-secret` pipeline parameter.
+See https://snyk.io/product/snyk-code/ and https://snyk.io/ for more information about the snyk tool.
```
Comment (simonbaird) [inline task/sast-snyk-check-oci-ta/0.1/README.md:9]: Why so many description changes? Will you back-port them to the old synk task?

Comment (lcarva) [reply task/sast-snyk-check-oci-ta/0.1/README.md:9]: I'm having these being generated by the `./hack/generate-readme.sh` script. I also noticed that a lot of the information removed seems outdated.

**Review by simonbaird — APPROVED** [review]

**Pull request merged**